scripts/setup-ros-workspace.bash

```diff
@@ -1,135 +1,309 @@
+  if [[ ${base_ws} != "<current>" ]]; then
+    if [[ $ros_version == 1 ]]; then
+      source $FRAMEWORK_BASE_PATH/${base_ws}_ros_ws_${ros_distro}/devel/setup.bash
+    else
+      source $FRAMEWORK_BASE_PATH/${base_ws}_ros_ws_${ros_distro}/install/setup.bash
+    fi
+  fi
+
+  mkdir -p ~/${ws_folder}/${ws_full_name}
+  cd ~/${ws_folder}/${ws_full_name} || print_and_exit "Could not change dir to workspace folder. Something went wrong."
+
+  if [ "$use_docker" = false ]; then # only build if not in docker, to avoide wrong dependencies
+    if [[ $ros_version == 1 ]]; then
+      wstool init src
+      catkin config -DCMAKE_BUILD_TYPE=RelwithDebInfo
+      catkin build
+    elif [[ $ros_version == 2 ]]; then
+      mkdir src
+    #   cb
+      colcon build --symlink-install --cmake-args -DCMAKE_BUILD_TYPE=RelWithDebInfo
+    fi
+  fi
+
+  # Go to Home folder of the user
+  cd || print_and_exit "Could not change back to home dir. This should never happen..."
+}
+
+setup_ros_team_ws_file() {
+  if [ -z "$1" ]; then
+    print_and_exit "No ros_team_ws_file given. Cannot setup workspace defines."
+  fi
+  local ros_team_ws_file=$1
+  if [ -z "$2" ]; then
+    print_and_exit "Not specified if docker should be used. Cannot setup workspace defines."
+  fi
+  local use_docker=$2
+    if [ -z "$2" ]; then
+    print_and_exit "Not specified if ros_teamws_file is for docker. Cannot setup workspace defines."
+  fi
+  local is_docker_rtw_file=$3
+
+  # Create a function name
+  fun_name_suffix=${chosen_ros_distro}
+
+  if [ "$ros_ws_prefix" != "-" ]; then
+    fun_name_suffix=${fun_name_suffix}_${ros_ws_prefix}
+  fi
+  if [ "$ros_ws_suffix" != "-" ]; then
+    fun_name_suffix=${fun_name_suffix}_${ros_ws_suffix}
+  fi
+
+  alias_name=_${ws_full_name}
+  fun_name="RosTeamWS_setup_${fun_name_suffix}"
+
+  cp "$ros_team_ws_file" "$ros_team_ws_file".bkp
+  # Comment out the old configuration if such exists - this is hard if using functions...
+  sed -i -e '/'"$fun_name"'/ s/^#*/OLD_/' "$ros_team_ws_file"
+  sed -i -e '/alias st_ros'"$ros_version=$fun_name"'/ s/^#*/#/' "$ros_team_ws_file"
+
+  if [ "$use_docker" = true ]; then
+    docker_image_tag=$(sed "s/[\/]/-/g" <(echo "ros-team-ws_${ubuntu_version_tag}__${ws_folder}__${ws_full_name}"))
+    docker_host_name="rtw-${ws_full_name}-docker"
+  else
+    docker_image_tag="-"
+    docker_host_name="-"
+  fi
+
+  local docker_support="$use_docker"
+  if [ "$is_docker_rtw_file" = true ]; then
+    source_path_rtw="  source /opt/RosTeamWS/ros_ws_$chosen_ros_distro/src/ros_team_workspace/scripts/environment/setup.bash \"\$RosTeamWS_DISTRO\" \"\$RosTeamWS_WS_FOLDER\" \"\$RosTeamWS_WS_PREFIX\" \"\$RosTeamWS_WS_SUFFIX\""
+    docker_support=false # don't use docker in docker
+  else
+    source_path_rtw="  source $FRAMEWORK_BASE_PATH/ros_ws_\"\$RosTeamWS_DISTRO\"/src/$FRAMEWORK_NAME/scripts/environment/setup.bash \"\$RosTeamWS_DISTRO\" \"\$RosTeamWS_WS_FOLDER\" \"\$RosTeamWS_WS_PREFIX\" \"\$RosTeamWS_WS_SUFFIX\""
```

Comment:
I suspect that this line here generates something very strange. The issue is that the genrated path get prefix `/opt/RosTeamWS/ros_ws_foxy/src/ros_team_workspace`. I have checked my environment and somewhere is `$FRAMEWORK_BASE_PATH` set wrongly.

Reply:
should be fixed